otherlibs/systhreads/st_stubs.c

```diff
@@ -666,6 +689,7 @@ static st_retcode create_tick_thread(void)
 CAMLprim value caml_thread_new(value clos)
 {
   CAMLparam1(clos);
+  CAMLlocal1(res);
```

Comment:
It is difficult for me to tell why you decided to root this variable, and I find it isky to root a variable that is, at various points within the function, not a valid OCaml value. Why are we doing this? (Can we have a comment that explains it, if it is subtle?)